common/src/main/kotlin/org/quizfight/common/Connection.kt

```diff
@@ -31,7 +34,7 @@ class SocketConnection(
     }
 
     private fun receiveAsync() = GlobalScope.launch {
-        while (socket.isConnected) {
+        while (!socket.isClosed && handleMessages) {
```

Comment:
`isConnected` will always be true after the Socket is initially connected, even if the socket is closed. This will usually happen on Socket creation unless you specifically use an unconnected Socket. If you do, I think it's fair to say that it needs to be connected before creating the `Connection` instance.